R/convertToEGRET.R

```diff
@@ -0,0 +1,212 @@
+    sample_df <- convertToEGRETSample(intdat, meta, qconvert)
+  } else {
+    sample_df <- convertToEGRETSample(intdat, meta, qconvert, daily_df)
+  }
+  
+  eList <- as.egret(INFO=info_df, Daily=daily_df, Sample=sample_df)
+  return(eList)
+}
+
+#' Convert the interpolation data.frame into the EGRET Sample dataframe.
+#'
+#' @param intdat data.frame of interpolation data
+#' @param meta loadflex metadata object; it must include constituent,
+#' flow, dates, conc.units, custom (station abbreviation: sta.abbr, and a short
+#' name for the constituent: consti.name)
+#' @param qconvert numeric conversion factor to get flow into cubic meters per second. Default
+#' conversion factor is for cubic feet per second.
+#' @param dailydat an EGRET Daily data.frame of flow values
+#' 
+#' @importFrom dplyr rename_
+#' @importFrom dplyr select
+#' @importFrom dplyr mutate
+#' @importFrom EGRET populateSampleColumns
+#' @importFrom dplyr left_join
+#' @importFrom dplyr bind_cols
+convertToEGRETSample <- function(intdat, meta, qconvert = 35.314667, dailydat = NULL) {
+  if(any(is.null(intdat), is.null(meta))) {
+    return(NA)
+  }
+  
+  flow_col <- verify_meta(meta, 'flow')
+  date_col <- verify_meta(meta, 'dates')
+  constituent <- verify_meta(meta, 'constituent')
+  sample_df1 <- intdat %>% 
+    rename_("value" = constituent,
+            "dateTime" = date_col)  %>%
+    select(dateTime, ConcHigh = value) %>% 
+    mutate(ConcLow = ConcHigh, 
+           Uncen=as.numeric(ConcHigh == ConcLow)) %>% 
+    populateSampleColumns() %>% 
+    mutate(dateTime = intdat[[date_col]],
+           Date = as.Date(Date))
+  
+  flow_data <- flowCorrectionEGRET(flowdat = intdat, 
+                                   flow.colname = flow_col,
+                                   date.colname = date_col,
+                                   qconvert = qconvert) %>% 
+    select(Date, Q, dateTime)
+  
+  sample_df <- sample_df1 %>%
+    left_join(flow_data, by=c("dateTime","Date"))
+  
+  if(!is.null(dailydat)){
+    subDaily <- select(sample_df, dateTime) %>%
+      left_join(select(dailydat, dateTime, SE, yHat), by="dateTime") %>%
+      mutate(ConcHat = yHat*exp((SE^2)/2)) 
+    
+    sample_df <- bind_cols(sample_df, subDaily)
+  }
+  
+  return(sample_df)
+}
+
+#' Convert a loadflex metadata object into the EGRET INFO dataframe.
+#' 
+#' @param meta loadflex metadata object; it must include constituent,
+#' flow, dates, conc.units, custom (station abbreviation: sta.abbr, and a short
+#' name for the constituent: consti.name)
+#' @param preds.type character specifying if the predictions being used are
+#' concentrations ("Conc") or fluxes ("Flux").
+#' 
+convertToEGRETInfo <- function(meta, preds.type = 'Conc') {
+  if(is.null(meta)) {
+    stop("metadata is required to create an EGRET eList")
+  }
+  
+  info_df <- data.frame(shortName=verify_meta(meta, 'station'),
+                        paramShortName=verify_meta(meta, c('custom', 'consti.name')),
```

Comment:
Yeah, sorry, I was trying to put it in it's own field.  I think I have it built in now though on its own re #121